scripts/cnode-helper-scripts/env

```diff
@@ -817,7 +817,7 @@ slotInterval() {
 # 2 : CLI - node response not valid json
 # 3 : Koios - general error
 getProtocolParams() {
-  if [[ -n ${KOIOS_API} ]]; then
+  if [[ -n ${KOIOS_API} ]] && [[ ${ENABLE_KOIOS} != N ]]; then
```

Comment:
For this one, it is preferable to use test_koios in cncli.sh like CNTools does [here](https://github.com/cardano-community/guild-operators/blob/1bc41b7487bcddc7c4e1e7dd268d9c0b63bbb205/scripts/cnode-helper-scripts/cntools.sh#L141) , so that it will test connectivity alongwith just variable definition

Reply:
test_koios only does this when **ENABLE_KOIOS = Y**, giving the SPO the control to disable KOIOS:
```
test_koios() {
  # make sure KOIOS_API is reachable, else fall back to cli
  [[ ${ENABLE_KOIOS} = 'Y' && -n ${KOIOS_API} && $(curl -sfk -o /dev/null -w "%{http_code}" -m 5 ${KOIOS_API}/tip | awk '{print $1}') = "200" ]] || unset KOIOS_API
}
```
I don't recall why I didn't make it an identical check of `= Y` instead of `!= N`. I remember I had something in mind when I did it, but maybe it was just an edge case of SPO typo/using `YES` if re-enabling it KOIOS without commenting out their setting.

I'm good with either check but prefer a consistent behavior when the SPO explicitly disables KOIOS via **ENABLE_KOIOS=N**. This way tools honor the SPO configuration, and no traffic would be observed via tcpdump or result in alerts for EDR systems.

Reply:
I may have misunderstood your statement as "don't do this". Did you actually mean:
```
if [[ -n ${KOIOS_API} ]] && [[ ! test_koios ]]; then
```

Reply:
No, as part of cncli.sh cncliInit function, we should call test_koios function just as CNTools does. This checks both ENABLE_KOIOS and that koios api is actually reachable and network supported. Else it will unset KOIOS_API variable. 

So if this is test function is called, `-n ${KOIOS_API}` should be enough to use everywhere after this. 

I can also see that we have USE_KOIOS_API in cncli.sh. We should remove this and replace with same logic.

Reply:
Gotcha, I'll modify the PR to do it during the init.

Reply:
Preferred solution:

1. revert env change
2. global search on `USE_KOIOS_API` in cncli.sh and remove where needed and replace with `-n ${KOIOS_API}` where check is done.
3. add a test_koios call at the appropriate place in `cncliInit` function, ie after loading/updating env but before needing to use KOIOS_API variable.